server/gateway.go

```diff
@@ -560,34 +562,34 @@ func (s *Server) solicitGateway(cfg *gatewayCfg) {
 	const connFmt = "Connecting to %s gateway %q (%s) at %s (attempt %v)"
 	const connErrFmt = "Error connecting to %s gateway %q (%s) at %s (attempt %v): %v"
 
-	for s.isRunning() && len(urls) > 0 {
+	for s.isRunning() && numURLs > 0 {
 		attempts++
 		report := shouldReportConnectErr(opts.ConnectionErrorReportAttempts, attempts)
-		// Iteration is random
-		for _, u := range urls {
-			address, err := s.getRandomIP(s.gateway.resolver, u.Host)
-			if err != nil {
-				s.Errorf("Error getting IP for %s gateway %q (%s): %v", typeStr, cfg.Name, u.Host, err)
-				continue
-			}
-			s.Debugf(connFmt, typeStr, cfg.Name, u.Host, address, attempts)
-			if report {
-				s.Noticef(connFmt, typeStr, cfg.Name, u.Host, address, attempts)
-			}
-			conn, err := net.DialTimeout("tcp", address, DEFAULT_ROUTE_DIAL)
-			if err == nil {
-				// We could connect, create the gateway connection and return.
-				s.createGateway(cfg, u, conn)
-				return
-			}
-			s.Debugf(connErrFmt, typeStr, cfg.Name, u.Host, address, attempts, err)
-			if report {
-				s.Errorf(connErrFmt, typeStr, cfg.Name, u.Host, address, attempts, err)
-			}
-			// Break this loop if server is being shutdown...
-			if !s.isRunning() {
-				break
-			}
+
+		// Pick a random url from the array.
```

Comment:
Where do we switch to an array from a map?

Reply:
ok I see, we do range from func below then also do random. I doubt these would be big, I might have opted for the shuffle command.